src/main/java/io/github/fablabsmc/fablabs/api/fiber/v1/serialization/JanksonValueSerializer.java

```diff
@@ -0,0 +1,154 @@
+	private final boolean minify;
+	private final Jankson jankson;
+
+	public JanksonValueSerializer(boolean minify) {
+		this(minify, Jankson.builder().build());
+	}
+
+	public JanksonValueSerializer(boolean minify, Jankson jankson) {
+		this.minify = minify;
+		this.jankson = jankson;
+	}
+
+	@Override
+	public JsonElement serializeBoolean(boolean value) {
+		return value ? JsonPrimitive.TRUE : JsonPrimitive.FALSE;
+	}
+
+	@Override
+	public boolean deserializeBoolean(JsonElement elem) throws ValueDeserializationException {
+		if (elem instanceof JsonPrimitive) {
+			Object value = ((JsonPrimitive) elem).getValue();
+
+			if (value instanceof Boolean) {
+				return (boolean) value;
+			}
+
+			throw new ValueDeserializationException(value, boolean.class, "JsonPrimitive not a boolean instance");
+		}
+
+		throw new ValueDeserializationException(elem, boolean.class, "JsonElement of wrong type");
+	}
+
+	@Override
+	public JsonElement serializeNumber(BigDecimal value) {
+		return new JsonPrimitive(value);
+	}
+
+	@Override
+	public BigDecimal deserializeNumber(JsonElement elem) throws ValueDeserializationException {
+		if (elem instanceof JsonPrimitive) {
+			String value = ((JsonPrimitive) elem).asString();
+
+			try {
+				return new BigDecimal(value);
+			} catch (NumberFormatException e) {
+				throw new ValueDeserializationException(value, BigDecimal.class, "JsonPrimitive string not a valid BigDecimal");
+			}
+		}
+
+		throw new ValueDeserializationException(elem, BigDecimal.class, "JsonElement of wrong type");
+	}
+
+	@Override
+	public JsonElement serializeString(String value) {
+		return new JsonPrimitive(value);
+	}
+
+	@Override
+	public String deserializeString(JsonElement elem) throws ValueDeserializationException {
+		if (elem instanceof JsonPrimitive) {
+			return ((JsonPrimitive) elem).asString();
+		}
+
+		throw new ValueDeserializationException(elem, String.class, "JsonElement of wrong type");
+	}
+
+	@Override
+	public JsonElement serializeList(List<JsonElement> value) {
+		JsonArray arr = new JsonArray();
+		arr.addAll(value);
+		return arr;
+	}
+
+	@Override
+	public List<JsonElement> deserializeList(JsonElement elem) throws ValueDeserializationException {
+		if (elem instanceof JsonArray) {
+			return ((JsonArray) elem);
+		}
+
+		throw new ValueDeserializationException(elem, List.class, "JsonElement of wrong type");
+	}
+
+	@Override
+	public JsonElement serializeMap(Map<String, JsonElement> value) {
+		JsonObject obj = new JsonObject();
+
+		for (Map.Entry<String, JsonElement> entry : value.entrySet()) {
+			obj.put(entry.getKey(), entry.getValue());
+		}
+
+		return obj;
+	}
+
+	@Override
+	public Map<String, JsonElement> deserializeMap(JsonElement elem) throws ValueDeserializationException {
+		if (elem instanceof JsonObject) {
+			return ((JsonObject) elem);
+		}
+
+		throw new ValueDeserializationException(elem, Map.class, "JsonElement of wrong type");
```

Comment:
I think we want to preserve ordering here, to avoid changing the file when saving. (ie. use `LinkedHashMap`)